kotlinc doesn't show errors from stderr

<!--
  This is the template for reporting ALE bugs. Make sure you try updating ALE
  to a more recent version before reporting a bug. Look through existing bug
  reports for similar issues before reporting a new one. Don't leave comments
  about new bugs in the comment section for old issues.

  Make sure to try disabling other plugins and trying to repeat your bug before
  reporting it in ALE. Some times problems can arise when two plugins are used
  together, but often your issues might be problems with other plugins.
-->

## Information

**VIM version**

NVIM v0.4.3
Build type: Release
<!-- Paste just the first two lines of :version here. -->

Operating System: Arch Linux, Linux Kernel 5.5.4<!-- Describe your operating system version. -->

## What went wrong

I was trying to use `kotlinc` to lint kotlin files and noticed it wasn't showing any lints, even from the following file `hello.kt`

```kotlin
fun main(args: Array<String>) {
	bad
}
```

Running essentially the same command listed in `:ALEInfo` `/bin/zsh -c 'kotlinc "/home/lovecraft/hello.kt"'` returns

```text
/home/lovecraft/hello.kt:2:2: error: unresolved reference: bad
	bad
 ^

```

I think the problem stems from `kotlinc` outputting to stderr instead of stdout. My hacky way of testing this was to move `kotlinc` to `kotlinc.orig` and create a new `kotlinc` that redirects stderr to stdout like below

```bash
#!/usr/bin/bash

kotlinc.orig "$@" 2>&1
```

After that ALE correctly displays the error in `hello.kt`. Hopefully any of this is helpful, I really appreciate all the work the ALE maintainers have put into such a great plugin!

<!-- Describe what went wrong here. Be specific. -->

## Reproducing the bug

<!-- Write a list of steps below. -->

1. Install `kotlinc`
2. Create a file with the same contents as `hello.kt`
3. Run ALE and see there are no errors shown

### :ALEInfo

```text
 Current Filetype: kotlin
Available Linters: ['kotlinc', 'ktlint', 'languageserver']
  Enabled Linters: ['kotlinc', 'ktlint', 'languageserver']
 Suggested Fixers: 
  'remove_trailing_lines' - Remove all blank lines at the end of a file.
  'trim_whitespace' - Remove all trailing whitespace characters at the end of every line.
 Linter Variables:

let g:ale_kotlin_kotlinc_classpath = ''
let g:ale_kotlin_kotlinc_config_file = '.ale_kotlinc_config'
let g:ale_kotlin_kotlinc_enable_config = 0
let g:ale_kotlin_kotlinc_module_filename = 'module.xml'
let g:ale_kotlin_kotlinc_options = ''
let g:ale_kotlin_kotlinc_sourcepath = ''
let g:ale_kotlin_kotlinc_use_module_file = 0
let g:ale_kotlin_languageserver_executable = 'kotlin-language-server'
 Global Variables:

let g:ale_cache_executable_check_failures = v:null
let g:ale_change_sign_column_color = 0
let g:ale_command_wrapper = ''
let g:ale_completion_delay = v:null
let g:ale_completion_enabled = 0
let g:ale_completion_max_suggestions = v:null
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'E'
let g:ale_echo_msg_format = '[%linter%] [%severity%] %code:%%s'
let g:ale_echo_msg_info_str = 'Info'
let g:ale_echo_msg_warning_str = 'W'
let g:ale_enabled = 1
let g:ale_fix_on_save = 0
let g:ale_fixers = {'c': ['clang-format'], 'rust': ['rustfmt'], 'cpp': ['clang-format'], 'python': ['black']}
let g:ale_history_enabled = 1
let g:ale_history_log_output = 1
let g:ale_keep_list_window_open = 0
let g:ale_lint_delay = 1000
let g:ale_lint_on_enter = 1
let g:ale_lint_on_filetype_changed = 1
let g:ale_lint_on_insert_leave = 1
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'normal'
let g:ale_linter_aliases = {}
let g:ale_linters = {}
let g:ale_linters_explicit = 0
let g:ale_list_vertical = 0
let g:ale_list_window_size = 10
let g:ale_loclist_msg_format = '[%linter%] [%severity%] %code:%%s'
let g:ale_lsp_root = {}
let g:ale_max_buffer_history_size = 20
let g:ale_max_signs = -1
let g:ale_maximum_file_size = v:null
let g:ale_open_list = 0
let g:ale_pattern_options = v:null
let g:ale_pattern_options_enabled = v:null
let g:ale_set_balloons = 0
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 0
let g:ale_set_signs = 1
let g:ale_sign_column_always = 1
let g:ale_sign_error = 'XX'
let g:ale_sign_info = '!!'
let g:ale_sign_offset = 1000000
let g:ale_sign_style_error = 'XX'
let g:ale_sign_style_warning = '!!'
let g:ale_sign_warning = '!!'
let g:ale_sign_highlight_linenrs = 0
let g:ale_statusline_format = v:null
let g:ale_type_map = {}
let g:ale_use_global_executables = v:null
let g:ale_virtualtext_cursor = 0
let g:ale_warn_about_trailing_blank_lines = 1
let g:ale_warn_about_trailing_whitespace = 1
  Command History:

(executable check - success) kotlinc
(finished - exit code 1) ['/bin/zsh', '-c', 'kotlinc    ''/home/lovecraft/hello.kt''']

<<<NO OUTPUT RETURNED>>>

(executable check - failure) ktlint
```
<!-- Paste the output of :ALEInfo here. Try :ALEInfoToClipboard -->
<!-- Make sure to run :ALEInfo from the buffer where the bug occurred. -->
<!-- Read the output. You might figure out what went wrong yourself. -->

--- a/ale_linters/kotlin/kotlinc.vim
+++ b/ale_linters/kotlin/kotlinc.vim
@@ -174,6 +174,7 @@ endfunction
 call ale#linter#Define('kotlin', {
 \   'name': 'kotlinc',
 \   'executable': 'kotlinc',
+\   'output_stream': 'stderr',
 \   'command': function('ale_linters#kotlin#kotlinc#RunWithImportPaths'),
 \   'callback': 'ale_linters#kotlin#kotlinc#Handle',
 \   'lint_file': 1,

